adam-core/src/main/scala/org/bdgenomics/adam/rdd/ShuffleRegionJoin.scala

```diff
@@ -17,16 +17,23 @@
  */
 package org.bdgenomics.adam.rdd
 
-import org.apache.spark.Partitioner
 import org.apache.spark.SparkContext._
+import org.apache.spark.{ Partitioner, SparkContext }
 import org.apache.spark.rdd.RDD
 import org.bdgenomics.adam.models.ReferenceRegion._
 import org.bdgenomics.adam.models.{ SequenceDictionary, ReferenceRegion }
 import scala.collection.mutable.ListBuffer
-import scala.math._
 import scala.reflect.ClassTag
 
-private[rdd] case class ShuffleRegionJoin(sd: SequenceDictionary, partitionSize: Long) extends RegionJoin {
+sealed trait ShuffleRegionJoin[T, U, RT, RU] extends RegionJoin[T, U, RT, RU] {
+
+  val sd: SequenceDictionary
+  val partitionSize: Long
+  @transient val sc: SparkContext
+
+  // Create the set of bins across the genome for parallel processing
+  protected val seqLengths = Map(sd.records.toSeq.map(rec => (rec.name, rec.length)): _*)
```

Comment:
This throws an error downstream in GenomeBins because it tries to set seqLengths from sd when sd is not yet set (is null). What would be the cleanest way to ensure this doesn't happen?